src/app/screens/transactionRequest/index.tsx

```diff
@@ -146,48 +191,14 @@ function TransactionRequest() {
           },
         });
       }
-    }
-    setHasSwitchedAccount(true);
-  };
-
-  const handleTxSigningRequest = async () => {
-    if (payload.txType === 'contract_call') {
-      await handleContractCallRequest(payload);
-    } else if (payload.txType === 'smart_contract') {
-      await handleContractDeployRequest(payload);
     } else {
-      navigate('/confirm-stx-tx', {
-        state: {
-          unsignedTx: payload.txHex,
-          sponsored: payload.sponsored,
-          isBrowserTx: true,
-          tabId,
-          requestToken,
-        },
-      });
-    }
-  };
-
-  const createRequestTx = async () => {
-    if (hasSwitchedAccount) {
-      if (!payload.txHex) {
-        if (payload.txType === 'token_transfer') {
-          await handleTokenTransferRequest(payload);
-        } else if (payload.txType === 'contract_call') {
-          await handleContractCallRequest(payload);
-        } else if (payload.txType === 'smart_contract') {
-          await handleContractDeployRequest(payload);
-        }
-      } else {
-        await handleTxSigningRequest();
-      }
+      await createRequestTx(selectedAccount!);
```

Comment:
Do we want to add a check if `selectedAccount` is not undefined here instead of using `!` symbol to make it safer?